lib/rand.c

```diff
@@ -267,3 +269,36 @@ CURLcode Curl_rand_hex(struct Curl_easy *data, unsigned char *rnd,
 
   return result;
 }
+
+/*
+ * Curl_rand_alnum() fills the 'rnd' buffer with a given 'num' size with random
+ * alphanumerical chars PLUS a null-terminating byte.
+ */
+
+CURLcode Curl_rand_alnum(struct Curl_easy *data, unsigned char *rnd,
+                         size_t num)
+{
+  CURLcode result = CURLE_OK;
+  const char alnum[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
```

Comment:
A microoptimization: use the same characters as base64encdec so the linker can save space and eliminate one copy. Assuming the extra three characters are allowed, of course. Also, this could be static const.

Reply:
@dfandrich While the extra chars might work in some context, there is no knowing where this function might end up being used. To remain useful I'd prefer sticking to plain alnum. I'll make it array that is not 0 terminated to fix this. Then the compiled can merge the arrays since the prefix is the same.